packages/components/src/data-model/word-query/word-query-resolver.js

```diff
@@ -0,0 +1,82 @@
+import { makeVar } from '@apollo/client/core'
+import { Logger, Digest } from 'alpheios-data-models'
+import WordQuery from '@comp/data-model/word-query/word-query.js'
+import WordQueryResult from '@comp/data-model/word-query/word-query-result.js'
+
+/**
+ * A class that would receive GraphQL requests and resolve them with data.
+ * Creates an instance of WordQuery for the query resolution.
+ */
+export default class WordQueryResolver {
+  constructor ({ getLexiconsFn, getShortLexiconsFn } = {}) {
+    this._getLexiconsFn = getLexiconsFn
+    this._getShortLexiconsFn = getShortLexiconsFn
+  }
+
+  // This function will be called by Apollo when new GraphQL query will arrive.
+  resolve (_, { storage, variables }) {
+    // This is a unique key of each request. We'll use its hash value to cache requests already resolved.
+    const key = Digest.fromObject({
+      word: variables.word,
+      language: variables.language,
+      contextForward: variables.contextForward,
+      contextBackward: variables.contextBackward
+    })
+    // The words map will store references to the results stored in the Apollo cache
+    if (!storage.words) { storage.words = new Map() }
+    if (!storage.words.has(key)) {
+      const wqResultWatcher = {
+        // This function will track changes in the top-level props of the result object
+        set: function (obj, prop, value) {
+          let wordReactive = storage.words.get(key) // eslint-disable-line prefer-const
```

Comment:
This should not happen right now because the function is used after the key is set (https://github.com/alpheios-project/alpheios-core/blob/dev-wordlist-refactoring/packages/components/src/data-model/word-query/word-query-resolver.js#L63-L65), but I agree it is not reliable enough to rely on that. Things may change in the future. I'll add the guard.